### PR TITLE
feat(response-rewrite): support brotli when use filters.regex option

### PR DESCRIPTION
**Pull request by yuweizzz**
### Description

Related: #10637 and #10562 

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/response-rewrite.lua [hunk 1]
```diff
@@ -294,6 +347,12 @@ function _M.body_filter(conf, ctx)
                 core.log.error("filters may not work as expected, inflate gzip err: ", err)
                 return
             end
+        elseif ctx.response_encoding == "br" and is_loaded then
```
Comment (shreemaan-abhishek) [inline apisix/plugins/response-rewrite.lua:350]: please choose a better name for `is_loaded`.

File: t/plugin/response-rewrite3.t [hunk 2]
```diff
@@ -53,6 +53,45 @@ add_block_preprocessor(sub {
...
+                local conf = {
+                    comp_level = 6,
+                    http_version = 1.1,
+                    lgblock = 0,
+                    lgwin = 19,
+                    min_length = 1,
+                    mode = 0,
+                    types = "*",
+                }
+                local brotli = require("apisix.plugins.brotli")
```
Comment (shreemaan-abhishek) [inline t/plugin/response-rewrite3.t:76]: if I am not wrong, you are importing this because nginx doesn't come with brotli by default.

Comment (yuweizzz) [reply t/plugin/response-rewrite3.t:76]: yes, in this part we use the functions from `brotli` plugin.

File: apisix/plugins/response-rewrite.lua [hunk 3]
```diff
@@ -227,6 +228,58 @@ local function inflate_gzip(data)
 end
 
 
+local function brotli_stream_decode(read_inputs, write_outputs)
+    -- read 64k data per times
```
[the remark this answers is not in the document]
Comment (yuweizzz) [reply apisix/plugins/response-rewrite.lua:232]: In `brotli-ffi` module, it uses `32 * 1024` as data size in streams operate,  so we use `64 * 1024` here to make a high efficiency.

**Comment by monkeyDluffy6017** [comment]
please make the ci pass

**Comment by yuweizzz** [comment]
I've reviewed the ci logs and the failure was due to a network error, could you please rerun the ci?

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**